tests/codebuild/run_integtest.sh

```diff
@@ -78,40 +103,101 @@ else
-        echo "Skipping installation of CRDs and operator"
-    else
-        # Goto directory that holds the CRD  
-        pushd sagemaker-k8s-operator-install-scripts
-            # Since OPERATOR_AWS_SECRET_ACCESS_KEY and OPERATOR_AWS_ACCESS_KEY_ID defined in build spec, we will not create new user
-            ./setup_awscreds
-
-            echo "Deploying the operator"
-            kustomize build config/default | kubectl apply -f -
-        popd
-
-        echo "Waiting for controller pod to be Ready"
-        # Wait to increase chance that pod is ready
-        # TODO: Should upgrade kubectl to version that supports `kubectl wait pods --all`
-        sleep 60
-    fi 
+    pushd sagemaker-k8s-operator-install-scripts
+        # Since OPERATOR_AWS_SECRET_ACCESS_KEY and OPERATOR_AWS_ACCESS_KEY_ID defined in build spec, we will not create new user
+        ./setup_awscreds
+        path_to_installer=$(pwd)
+    popd
 popd
 
-# Run the integration test file
-cd tests/codebuild/ && ./run_all_sample_test.sh
+echo "RUN INTEGRATION TESTS FOR THE CLUSTER SCOPED OPERATOR DEPLOYMENT"
+# Allow for overriding the installation of the CRDs/controller image from the
+# build scripts if we want to use our own installation
+if [ "${SKIP_INSTALLATION}" == "true" ]; then
+    echo "Skipping installation of CRDs and operator"
+else
+    pushd sagemaker-k8s-operator/sagemaker-k8s-operator-install-scripts
+        echo "Deploying the operator to the default namespace"
+        kustomize build config/default | kubectl apply -f -
+    popd
+    echo "Waiting for controller pod to be Ready"
+    # Wait to increase chance that pod is ready
+    # TODO: Should upgrade kubectl to version that supports `kubectl wait pods --all`
+    sleep 60
+    kubectl get pods --all-namespaces | grep sagemaker
+fi 
+
+echo "Starting Integ Tests in default namespace"
+pushd tests/codebuild
+    ./run_all_sample_test.sh "default"
+popd
 
 echo "Skipping private link test"
 #cd private-link-test && ./run_private_link_integration_test "${cluster_name}" "us-west-2"
+
+echo "RUN INTEGRATION TESTS FOR THE NAMESPACED OPERATOR DEPLOYMENT"
+# HELPER FUNCTIONS
+function generate_iam_role_name {
+    local crd_namespace="$1"
+    local cluster=$(echo ${cluster_name} | cut -d'/' -f2)
+
+    role_name="${cluster}-${crd_namespace}"
+}
+
+function operator_namespace_deploy {
+    local crd_namespace="$1"
+
+    # Goto directory that holds the CRD 
+    pushd sagemaker-k8s-operator/sagemaker-k8s-operator-install-scripts
+        kustomize build config/crd | kubectl apply -f -
+        generate_namespace_operator_installer ${crd_namespace}
+        kubectl apply -f temp_file.yaml
+    popd
+    echo "Waiting for controller pod to be Ready"
+    # Wait to increase chance that pod is ready
+    # TODO: Should upgrade kubectl to version that supports `kubectl wait pods --all`
+    sleep 60
+    echo "Check that the manager pod is running"
+    kubectl get pods --all-namespaces | grep sagemaker
```

Comment:
Are we really checking anything here?

Reply:
probably you might want to run 
```
if [ $? eq 0] 
 success
else 
 fail
```

Reply:
This is a debugging output more than anything. The `sleep 60` ensures it's all up and running.

Reply:
This helped me debug a couple of times and hence decided to leave it in there. the sleep guarantees sufficient time if all is well, the log confirms it.

Reply:
Probably change the log message in line 168, thats a bit misleading.

Reply:
done